R/expr__expr.R

```diff
@@ -675,37 +651,52 @@ construct_ProtoExprArray = function(...) {
-#' in parallel R sessions. The communication speed between processes is quite slower than between
-#' threads. Will likely only give a speed-up in a "low IO - high CPU" usecase. A single map will not
-#' be paralleled, only in case of multiple `$map`(s) in the query these can be run in parallel.
+#' @param in_background Boolean. Whether to execute the map in a background R
+#' process. Combined with setting e.g. `pl$set_options(rpool_cap = 4)` it can speed
+#' up some slow R functions as they can run in parallel R sessions. The
+#' communication speed between processes is quite slower than between threads.
+#' This will likely only give a speed-up in a "low IO - high CPU" usecase.
+#' If there are multiple `$map(in_background = TRUE)` calls in the query, they
+#' will be run in parallel.
 #'
 #' @return Expr
-#' @details Sometime some specific R function is just necessary to perform a column transformation.
-#' Using R maps is slower than native polars. User function must take one polars `Series` as input
-#' and the return should be a `Series` or any Robj convertible into a `Series` (e.g. vectors).
-#' Map fully supports `browser()`. If `in_background = FALSE` the function can access any global
-#' variable of the R session. But all R maps in the query sequentially share the same main R
-#' session. Any native polars computations can still be executed meanwhile. In
-#' `in_background = TRUE` the map will run in one or more other R sessions and will not have access
-#' to global variables. Use `pl$set_options(rpool_cap = 4)` and `pl$options$rpool_cap` to see and
-#' view number of parallel R sessions.
+#' @details
+#' It is sometimes necessary to apply a specific R function on one or several
+#' columns. However, note that using R code in `$map()` is slower than native
+#' polars. The user function must take one polars `Series` as input and the return
+#' should be a `Series` or any Robj convertible into a `Series` (e.g. vectors).
+#' Map fully supports `browser()`.
+#'
+#' If `in_background = FALSE` the function can access any global variable of the
+#' R session. However, note that several calls to `$map()` will sequentially
+#' share the same main R session, so the global environment might change between
+#' the start of the query and the moment a `map()` call is evaluated. Any native
+#' polars computations can still be executed meanwhile. If `in_background = TRUE`,
+#' the map will run in one or more other R sessions and will not have access
+#' to global variables. Use `pl$set_options(rpool_cap = 4)` and `pl$options$rpool_cap`
+#' to see and view number of parallel R sessions.
+#'
 #' @name Expr_map
 #' @examples
-#' pl$DataFrame(iris)$select(pl$col("Sepal.Length")$map(\(x) {
-#'   paste("cheese", as.character(x$to_vector()))
-#' }, pl$dtypes$Utf8))
+#' pl$DataFrame(iris)$
+#'   select(
+#'     pl$col("Sepal.Length")$map(\(x) {
+#'       paste("cheese", as.character(x$to_vector()))
+#'     }, pl$dtypes$Utf8)
+#'    )
```

Comment:
Perhaps the indentation is misaligned?

Reply:
thanks, I ran `make fmt`, it should be fixed